rest-api/src/test/java/com/cloudbees/workflow/rest/endpoints/FlowNodeAPITest.java

```diff
@@ -94,6 +95,160 @@ public void test_success_flow() throws Exception {
         assert_log_ok(webClient, jsonReadWrite, workflowRuns);
     }
 
+    @Test
+    @Issue("JENKINS-33700")
+    public void test_unstable_flow() throws Exception {
+        // Succeeds but custom-marked as unstable
+        WorkflowJob passUnstable = jenkinsRule.jenkins.createProject(WorkflowJob.class, "SafelyUnstable");
+        // Error generated, but caught and custom-marked as unstable
+        WorkflowJob failUnstable = jenkinsRule.jenkins.createProject(WorkflowJob.class, "DangerouslyUnstable");
+
+        passUnstable.setDefinition(new CpsFlowDefinition("" +
+                "stage 'first'\n" +
+                "echo 'works'\n" +
+                "stage 'second'\n" +
+                "currentBuild.result = 'UNSTABLE'\n" +
+                "echo 'ran things'\n" +
+                "stage 'end'\n" +
+                "echo 'done'"
+                ));
+
+        failUnstable.setDefinition(new CpsFlowDefinition("" +
```

Comment:
Two separate scenarios should probably be in different test cases, so you can see from the report which is failing if any.